### DISCUSSION TITLE
xmake 尝试使用ndkbuild

### DISCUSSION THREAD

**Discussion by caxiejiang** [xmake-io/xmake#3011] [Q&A] [answered]
xmake 中尝试使用 ndk-build，并没有想象中进行编译(貌似什么都没有发生)
![image](https://user-images.githubusercontent.com/16971884/199407609-41a187b8-0956-4165-98d5-384a7ae43793.png)
当前目录下文件如下：
![image](https://user-images.githubusercontent.com/16971884/199407750-4bd822de-3fea-4b06-9799-bf55cfab96ca.png)
请问如此写有问题么

**Comment by waruqi** [comment, marked as answer]
进到 jni 上层目录执行下试试

**Comment by caxiejiang** [comment]
我尝试了进入jni上层目录，确实是可以正常编译生成动态库，但是奇怪的是生成的动态库只有6kb，而且看起来在过程中并没有执行源代码编译，而是直接生成动态库的
![image](https://user-images.githubusercontent.com/16971884/199416518-c61cd2dd-3940-4150-8914-017abf2cbb1a.png)
我的编译命令如下：
xmake f -p android --trybuild=ndkbuild
xmake

同时，我尝试了直接通过ndk-build运行同一个Android.mk脚本，可以明显看到有代码编译过程：
![image](https://user-images.githubusercontent.com/16971884/199418831-bca3ea75-e6a2-484d-a022-3d49737dccfe.png)

似乎看起来xmake运行还少了什么东西一样

附：
直接运行ndk-build編出來的库
![image](https://user-images.githubusercontent.com/16971884/199419331-aadbb1af-0e1b-40d7-9cc6-fb449a0b6be4.png)
xmake trybuild编译出来的库大小
![image](https://user-images.githubusercontent.com/16971884/199419535-2a4eaede-0e2f-44ca-9679-d10511ebd4b1.png)

这样的size差距我感觉不是很寻常，应该不是一些编译参数的问题（况且他们都是调用Andorid.mk）所以想请教您这块问题
万分感谢您的解答

**Reply by waruqi** [reply]
xmake 仅仅只是调用 ndkbuild 而已，没啥特别处理，你可以自己走 ndkbuild 跑下 应该一样的。

**Reply by caxiejiang** [reply]
问题应该找到。
Android.mk的上级目录执行xmake trybuild之后，编译时源码的路径找不到导致没有编译到cpp文件
感谢您的指导！
